dali/kernels/common/copy.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_COMMON_COPY_H_
+#define DALI_KERNELS_COMMON_COPY_H_
+
+#include <cuda_runtime.h>
+#include "dali/kernels/tensor_view.h"
+#include "dali/kernels/backend_tags.h"
+
+namespace dali {
+namespace kernels {
+
+template <typename Storage1, typename T1, int ndim1, typename Storage2, typename T2, int ndim2>
```

Comment:
I'd change the 1, 2 suffixes with Input, Output. E.g. `StorageInput`, `StorageOutput`